### PR TITLE
Closing #1223 - BeginDistribution / FinishDistribution Changes

### PR DESCRIPTION
**Pull request by xBalbinus**
Closes: #1223 

## Purpose of the change

Renames k.BeginDistribution(ctx, gauge) to k.moveUpcomingGaugeToActiveGauge, and k.FinishDistribution(ctx, gauge) to k.moveActiveGaugeToFinishedGauge(ctx, gauge)

## Brief change log

- Added function definitions to export_test.go
- Renamed the original functions
- Fixed function calls on tests to match the names of the current functions

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	github.com/CosmWasm/wasmd v0.24.0
-	github.com/cosmos/cosmos-sdk v0.45.3
+	github.com/cosmos/cosmos-sdk v0.45.4
```
Comment (p0mvn) [inline go.mod:7]: What is the reason for this? Is this necessary?

We usually upgrade our SDK fork in the replace directive here:
https://github.com/osmosis-labs/osmosis/blob/d8918936a898bcc982147f8e42d096a0f9a473db/go.mod#L271

**Review by alexanderbez — APPROVED** [review]

File: docs/core/proto-docs.md [hunk 2]
```diff
@@ -4,31 +4,6 @@
 
 ## Table of Contents
 
-- [osmosis/claim/v1beta1/claim.proto](#osmosis/claim/v1beta1/claim.proto)
```
Comment (p0mvn) [inline docs/core/proto-docs.md:7]: I don't see any protos updated in this PR. Why is this changed?

File: go.mod (hunk 1 above)
Comment (xBalbinus) [reply go.mod:7]: I'll change this back - I think I got this as a part of pulling from main before making these changes but I could be wrong
